docs/spec/auth/vesting.md

```diff
@@ -74,28 +58,46 @@ type BaseVestingAccount struct {
 
     OriginalVesting  Coins // coins in account upon initialization
     DelegatedFree    Coins // coins that are vested and delegated
-    EndTime          Time // when the coins become unlocked
+    DelegatedVesting Coins // coins that vesting and delegated
+
+    EndTime  Time // when the coins become unlocked
 }
 
 // ContinuousVestingAccount implements the VestingAccount interface. It
 // continuously vests by unlocking coins linearly with respect to time.
 type ContinuousVestingAccount struct {
-    BaseAccount
     BaseVestingAccount
 
-    DelegatedVesting Coins // coins that vesting and delegated
-    StartTime        Time // when the coins start to vest
+    StartTime  Time // when the coins start to vest
 }
 
 // DelayedVestingAccount implements the VestingAccount interface. It vests all
 // coins after a specific time, but non prior. In other words, it keeps them
 // locked until a specified time.
 type DelayedVestingAccount struct {
-    BaseAccount
     BaseVestingAccount
 }
 ```
 
+In order to facilitate less ad-hoc type checking and assertions and to support
+flexibility in account usage, the existing `Account` interface is updated to contain
+the following:
+
+```go
+type Account interface {
+    // ...
+
+    // Calculates the amount of coins that can be sent to other accounts given
+    // the current time.
+    SpendableCoins(Time) Coins
+
+    // Delegation and undelegation accounting that returns the resulting base
+    // coins amount.
+    TrackDelegation(Time, Coins)
```

Comment:
TrackDelegation/TrackUndelegation are weird names.  DelegateCoins/UndelegateCoins seems fine?

Reply:
Ehhh, but you're not actually delegating or undelegating. You're simply performing some accounting. I think `Track*` is just fine imho.